name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/XTBPolandPDFExtractor.java

```diff
@@ -0,0 +1,158 @@
+                                        //ACC, EUR 11:18:08 dot. ETF
+                                        // @formatter:on
+                                        section -> section //
+                                                        .attributes("shares", "tickerSymbol", "name", "tradeId", "date",
+                                                                        "amount", "currency", "time", "totalcost") //
+                                                        .match("^(\\d+) (?<tradeId>\\d+) (?<tickerSymbol>[A-Z0-9\\\\._-]{1,10}(?:\\\\.[A-Z]{1,4})?) (?<name>.*?) (?<shares>[\\.,'\\d]+) (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) \\w+ (?<amount>[\\.,'\\d]+) Prawa \\w+ (?<currency>[A-Z]{3}) ([\\.,'\\d]+) ([\\.,'\\d]+) ([\\.,'\\d]+) (?<totalcost>[\\.,'\\d]+).*$") //
+                                                        .match("^([A-Z]{3})\\, ([A-Z]{3}) (?<time>[\\d]{2}:[\\d]{2}:[\\d]{2}).*")
+                                                        .assign((t, v) -> {
+
+                                                            t.setDate(asDate(v.get("date"), v.get("time")));
+                                                            t.setShares(asShares(v.get("shares")));
+                                                            t.setAmount(asAmount(v.get("amount")));
+                                                            t.setCurrencyCode(v.get("currency"));
+                                                            t.setAmount(asAmount(v.get("amount")));
+                                                            var fees = Money.of(v.get("currency"),
+                                                                            asAmount(v.get("totalcost")));
+
+                                                            checkAndSetFee(fees, t, type.getCurrentContext());
+
+                                                            t.setSecurity(getOrCreateSecurity(v));
+                                                        })
+                        )
+
+                        .conclude(ExtractorUtils.fixGrossValueBuySell())
+                        .wrap(BuySellEntryItem::new);
+    }
+
+    @Override
+    protected long asShares(String value)
+    {
+        var newNumberFormat = (DecimalFormat) NumberFormat.getInstance(Locale.ENGLISH);
+        // Ensure the parser is strict, just in case (though not strictly
+        // necessary here)
+        newNumberFormat.setParseIntegerOnly(false);
+        double newNumberDouble = 0;
+        try
+        {
+            newNumberDouble = newNumberFormat.parse(value).doubleValue();
+        }
+        catch (ParseException e)
+        {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+            return 0;
+        }
+        value = trim(value).replaceAll("\\s", "");
+        
+        return Math.abs(Math.round(newNumberDouble * Values.Share.factor()));
+
+    }
```

Comment:
Look at the other importers like "ErsteBank" to solve this problem. 👍🏻 
--> ExtractorUtils.convertToNumberLong